io/src/macros.rs

```diff
@@ -24,15 +19,18 @@ macro_rules! impl_write {
     }
 }
 
-#[macro_export]
-/// Because we cannot provide a blanket implementation of [`std::io::Write`] for all implementers
-/// of this crate's `io::Write` trait, we provide this macro instead.
+/// Implements [`crate::Write`] for `$ty`.
 ///
-/// This macro will implement `Write` given a `write` and `flush` fn, either by implementing the
-/// crate's native `io::Write` trait directly, or a more generic trait from `std` for users using
-/// that feature. In any case, this crate's `io::Write` feature will be implemented for the given
-/// type, even if indirectly.
+/// Also implements `std::io::Write` for `$ty` if the `std` feature is enabled.
```

Comment:
Can't we doclink this as well?
```suggestion
/// Also implements [`std::io::Write`] for `$ty` if the `std` feature is enabled.
```

Reply:
In fbb1633a0faae3b826835a42e810b03bd907eead:

Also on this line: we should clarify that the `std` feature must be enabled *in `bitcoin-io`* rather than in the calling crate.